jwst/assign_wcs/util.py

```diff
@@ -471,3 +471,25 @@ def get_num_msa_open_shutters(shutter_state):
     if 'x' in shutter_state:
         num += 1
     return num
+
+
+def update_s_region(model):
```

Comment:
If this function is being called at the time of assign_wcs, when for example IFU images are still in their full 2D form containing many slices, will the wcs.footprint() method compute a reasonable result? Can it handle complex WCS objects like that?

Reply:
Ideally, for something like IFU data, it should compute the footprint of the physical IFU aperture on the sky.

Reply:
Good catch! It will work for the MIRI IFU but not for Nirspec (IFU, FS, or MSA).

Reply:
I excluded the complex NRS types for now, until the WCS is modified to run in a simplified way.